horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -108,29 +110,87 @@ public void testTransformationWithoutSchema(TestInfo info) throws InterruptedExc
       assertFalse(run.trashed);
       assertNewDataset(dataSetQueue, runId);
    }
-   private int labelValuesSetup(Test t) throws JsonProcessingException {
+   private int loadRun(String json,String uri,String testName) throws JsonProcessingException {
+      ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);;
+      JsonNode data = mapper.readTree(json);
+      String idString = uploadRun(data,testName,uri);
+      int id = Integer.parseInt(idString);
+      return id;
+   }
```

Comment:
Given that we are defining the same `loadRun` method in both `RunServiceTest` and `TestServiceTest` (and the implemenation looks the same), I would suggest to move the method in the `BaseServiceTest` to avoid duplication

Reply:
Yep, that one is probably good to move into the `BaseServiceTest` as an alternative form of `uploadRun`